tracer/test/test-applications/integrations/Samples.DataStreams.ManualAPI/Program.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace;
+
+namespace Samples.DataStreams.ManualAPI;
+
+public class Program
+{
+    private static ConcurrentQueue<string> _queue = new();
+
+    private static async Task Main(string[] args)
+    {
+        var writing = Send("my message");
+        var reading = Receive();
+
+        await Task.WhenAll(reading, writing);
+    }
+
+    private static async Task Send(string message)
+    {
+        using var scope = Tracer.Instance.StartActive("Samples.DataStreams.ManualAPI.Send");
+
+        await Task.Delay(millisecondsDelay: 100);
+
+        Console.WriteLine("Sending one message to the queue...");
+        var sb = new StringBuilder();
+        var injector = new SpanContextInjector();
+        injector.InjectIncludingDsm(sb, (b, k, v) => b.Append($"{k}:{v};"), scope.Span.Context, "ConcurrentQueue", nameof(_queue));
+        sb.Append(message);
+        _queue.Enqueue(sb.ToString());
+        Console.WriteLine("message sent");
+    }
+
+    private static async Task<string> Receive()
+    {
+        Console.WriteLine("Receiving one message from the queue...");
+        string result;
+        while (!_queue.TryDequeue(out result))
```

Comment:
Totally unnecessary to make any changes to this code, but just flagging `BlockingCollection<T>` in case you weren't aware of it 🙂  https://learn.microsoft.com/en-us/dotnet/standard/collections/thread-safe/blockingcollection-overview, as it would let you remove all the Task.Delay(), so might speed things up

Reply:
ok nice, didn't know about it, but I'll leave this sample code as it is :D